### PR TITLE
GSOC WME3D: 3d object selection and attachment

### PR DESCRIPTION
**Pull request by GunnarBirke**

### CODE REVIEW

File: engines/wintermute/base/gfx/x/modelx.cpp [hunk 1]
```diff
@@ -38,6 +38,8 @@
 #include "engines/wintermute/utils/path_util.h"
 #include "engines/wintermute/utils/utils.h"
 
+#include <limits.h>
```
Comment (aquadran) [inline engines/wintermute/base/gfx/x/modelx.cpp:41]: this might be not portable

File: engines/wintermute/base/gfx/x/modelx.cpp [hunk 2]
```diff
@@ -371,120 +373,77 @@ bool ModelX::isTransparentAt(int x, int y) {
...
-	if (customViewport) {
-		x += _gameRef->_renderer->_drawOffsetX;
-		y += _gameRef->_renderer->_drawOffsetY;
+	if (_rootFrame) {
+		_rootFrame->getBoundingBox(&_BBoxStart, &_BBoxEnd);
 	}
 
-	Math::Vector3d pickRayDir;
-	Math::Vector3d pickRayOrig;
+	_boundingRect.left = INT_MAX;
```
Comment (aquadran) [inline engines/wintermute/base/gfx/x/modelx.cpp:399]: same as above too limits.h

Comment (GunnarBirke) [reply engines/wintermute/base/gfx/x/modelx.cpp]: REPLACED it by INT_MAX_VALUE and new constant INT_MIN_VALUE. Assuming two complement is ok? (so INT_MIN_VALUE = -INT_MAX_VALUE -1)

File: engines/wintermute/base/gfx/x/modelx.cpp [hunk 3]
```diff
@@ -371,120 +372,77 @@ bool ModelX::isTransparentAt(int x, int y) {
...
+	_boundingRect.top = INT_MAX_VALUE;
+	_boundingRect.right = INT_MIN_VALUE;
+	_boundingRect.bottom = INT_MIN_VALUE;
 
-	// Compute the vector of the pick ray in screen space
-	Math::Vector3d vec;
-	vec.x() = (((2.0f * x) / _drawingViewport.width()) - 1) / _lastProjMat(0, 0);
-	vec.y() = -(((2.0f * y) / _drawingViewport.height()) - 1) / _lastProjMat(1, 1);
-	vec.z() = 1.0f;
+	Math::Vector3d vec2d(0,0,0);
```
Comment (aquadran) [inline engines/wintermute/base/gfx/x/modelx.cpp:403]: missing spaces

File: engines/wintermute/wintermute.h [hunk 4]
```diff
@@ -36,6 +36,7 @@ class DebuggerController;
 struct WMEGameDescription;
 
 const int INT_MAX_VALUE = 0x7fffffff;
+const int INT_MIN_VALUE = -INT_MAX_VALUE - 1;
```
Comment (aquadran) [inline engines/wintermute/wintermute.h:39]: add comment in the same line // ENABLE_WME3D

File: engines/wintermute/base/gfx/x/modelx.cpp (hunk 3 above)
Comment (GunnarBirke) [reply engines/wintermute/base/gfx/x/modelx.cpp]: Spaces added

File: engines/wintermute/wintermute.h (hunk 4 above)
Comment (GunnarBirke) [reply engines/wintermute/wintermute.h]: Comment added

**Comment by aquadran** [comment]
you have to resolve conflicts

**Pull request merged**